### ISSUE TITLE
MAPE loss not conveging with small values

### ISSUE THREAD

**Issue by PirashanthR** [catboost/catboost#281]
Problem: {Hi, 
I'm trying to use the catboost regressor in order to provide a solution to a data challenge.
I need to use the MAPE loss and I'm dealing with pretty small target values (<0.1 in most of the cases).
Using the Catboost Regressor without special tuning:

![image](https://user-images.githubusercontent.com/22860171/37874604-7c37a028-3032-11e8-8786-91c9f131dce7.png)

Gives me some divergent outcome: 

![image](https://user-images.githubusercontent.com/22860171/37874595-4a929b22-3032-11e8-8eac-866e613208fb.png)

I tried to multiply by a certain factor all my target values, this leads  to convergence but with a really bad accuracy (about 0.99 in mape score while I'm getting 0.22 with LGBM).  I tried to tune the learning rate and the number of iteration in order to improve the outcome in that case I could reach a score of 0.30 (setting the learning rate to 10000...) but it's far from LGBM outcome.

Do you have any solution that will allow me to use Catboost Regressor with the MAPE loss properly? 

Thank you

 }
catboost version: {0.6.3}
Operating System: {Google Colab}
CPU: {CPU}

**Comment by annaveronika** [comment]
could you share the dataset? Or part of it if it's reproducable on the part

**Comment by PirashanthR** [comment]
Thank you for your quick answer.
Here is an archive that contains part of the dataset I've used (10 000 out of 600 000 samples).
I've also added a script that allows running the basic tests that I did.

[test_mape_loss_catboost.zip](https://github.com/catboost/catboost/files/1849052/test_mape_loss_catboost.zip)

PS: Link to the full dataset: 
https://challengedata.ens.fr/en/challenge/34/volatility_prediction_in_financial_markets.html

**Comment by annaveronika** [comment]
If we understand correctly, the problem here is not with MAPE convergence. Standard learning rate is too big fot this dataset, you need do decrease learning rate to at least 1e-3, and you will get convergence, the score will be around 0.20, which is a bit better than one of LightGBM
